src/main/java/com/appsflyer/otelawsmetrics/OtelMetricPublisher.java

```diff
@@ -0,0 +1,289 @@
+
+                CoreMetric.TIME_TO_LAST_BYTE.name(), new MetricStrategyWithoutErrors(new DurationStrategy(meter,
+                        metricPrefix + "time_to_last_byte",
+                        "Elapsed time from sending the HTTP request (including acquiring a connection) to receiving the last byte of the response")),
+
+                CoreMetric.UNMARSHALLING_DURATION.name(), new MetricStrategyWithoutErrors(new DurationStrategy(meter,
+                        metricPrefix + "unmarshalling_duration",
+                        "The time it takes to unmarshall an HTTP response to an SDK response")));
+    }
+
+    private Map<String, MetricStrategy> initializeHttpStrategies(Meter meter) {
+        return Map.of(HttpMetric.AVAILABLE_CONCURRENCY.name(), new MetricStrategyWithoutErrors(new LongHistogramStrategy(meter,
+                        metricPrefix + "available_concurrency",
+                        "The number of remaining concurrent requests that can be supported by the HTTP client without needing to establish another connection")),
+
+                HttpMetric.CONCURRENCY_ACQUIRE_DURATION.name(), new MetricStrategyWithoutErrors(new DurationStrategy(meter,
+                        metricPrefix + "concurrency_acquire_duration",
+                        "The time taken to acquire a channel from the connection pool")),
+
+                HttpMetric.LEASED_CONCURRENCY.name(), new MetricStrategyWithoutErrors(new LongHistogramStrategy(meter,
+                        metricPrefix + "leased_concurrency",
+                        "The number of request currently being executed by the HTTP client")),
+
+                HttpMetric.MAX_CONCURRENCY.name(), new MetricStrategyWithoutErrors(new LongHistogramStrategy(meter,
+                        metricPrefix + "max_concurrency",
+                        "The max number of concurrent requests supported by the HTTP client")),
+
+                HttpMetric.PENDING_CONCURRENCY_ACQUIRES.name(), new MetricStrategyWithoutErrors(new LongHistogramStrategy(meter,
+                        metricPrefix + "pending_concurrency_acquires",
+                        "The number of requests that are blocked, waiting for another TCP connection or a new stream to be available from the connection pool")));
+    }
+
+    private void publishInternal(MetricCollection metricCollection) {
+        try {
+            // Start processing from the root per-request metrics
+            processPerRequestMetrics(metricCollection);
+        } catch (Exception e) {
+            log.error("An error occurred while publishing metrics", e);
+        }
+    }
+
+    private void recordMetrics(Map<String, MetricRecord<?>> metricsMap,
+                               Attributes attributes,
+                               Map<String, MetricStrategy> metricStrategies) {
+        for (Map.Entry<String, MetricStrategy> entry : metricStrategies.entrySet()) {
```

Comment:
Seeing as `HashMap.forEach` is just a fancy way to write `for (... : map.entrySet())`, I will prefer to avoid the extra allocation.  

![image](https://github.com/user-attachments/assets/7cab3fc6-eb18-465b-8975-1ba12a0e093c)